scripts/build/gen_isr_tables.py

```diff
@@ -26,6 +26,8 @@
 FIRST_LVL_INTERRUPTS = 0x000000FF
 SECND_LVL_INTERRUPTS = 0x0000FF00
 THIRD_LVL_INTERRUPTS = 0x00FF0000
+SECND_LVL_SHIFT = 8
```

Comment:
With this file, I think as long as you know the number of bits occupied at each level, the masks and shifts can be derived.

```py
FIRST_LVL_INTERRUPTS = 0
SCND_LVL_INTERRUPTS = 0
THRD_LVL_INTERRUPTS = 0

INTERRUPT_BITS = [8, 8, 8]

def update_masks():

    if sum(INTERRUPT_BITS) > 32:
      raise ValueError('too many interrupt bits')

    FIRST_LVL_INTERRUPTS = bit_mask(INTERRUPT_BITS[0])
    SCND_LVL_INTERRUPTS = bit_mask(INTERRUPT_BITS[1]) << INTERRUPT_BITS[0]
    THRD_LVL_INTERRUPTS = bit_mask(INTERRUPT_BITS[2]) << INTERRUPT_BITS[0] + INTERRUPT_BITS[1]
```

Could call that once after reading all three Kconfig values and it should be fine.

Reply:
Done!